aspnetcore/security/enforcing-ssl.md

```diff
@@ -78,46 +108,51 @@ The preceding highlighted code:
 * Sets [HttpsRedirectionOptions.RedirectStatusCode](/dotnet/api/microsoft.aspnetcore.httpspolicy.httpsredirectionoptions.redirectstatuscode) to [Status307TemporaryRedirect](/dotnet/api/microsoft.aspnetcore.http.statuscodes.status307temporaryredirect), which is the default value. Use the fields of the [StatusCodes](/dotnet/api/microsoft.aspnetcore.http.statuscodes) class for assignments to `RedirectStatusCode`.
 * Sets the HTTPS port to 5001. The default value is 443.
 
-The following mechanisms set the port automatically:
-
-* The middleware can discover the ports via [IServerAddressesFeature](/dotnet/api/microsoft.aspnetcore.hosting.server.features.iserveraddressesfeature) when the following conditions apply:
-
-  * Kestrel or HTTP.sys is used directly with HTTPS endpoints (also applies to running the app with Visual Studio Code's debugger).
-  * Only **one HTTPS port** is used by the app.
+#### Configure permanent redirects in production
 
-* Visual Studio is used:
+The middleware defaults to sending a [Status307TemporaryRedirect](/dotnet/api/microsoft.aspnetcore.http.statuscodes.status307temporaryredirect) with all redirects. If you prefer to send a permanent redirect status code when the app is in a non-Development environment, wrap the middleware options configuration in a conditional check for a non-Development environment.
 
-  * IIS Express has HTTPS enabled.
-  * *launchSettings.json* sets the `sslPort` for IIS Express.
+When configuring an `IWebHostBuilder` in *Startup.cs*:
 
-> [!NOTE]
-> When an app is run behind a reverse proxy (for example, IIS, IIS Express), `IServerAddressesFeature` isn't available. The port must be manually configured. When the port isn't set, requests aren't redirected.
-
-The port can be configured by setting the [https_port Web Host configuration setting](xref:fundamentals/host/web-host#https-port):
+```csharp
+public void ConfigureServices(IServiceCollection services)
+{
+    // IHostingEnvironment (stored in _env) is injected into the Startup class.
+    if (!_env.IsDevelopment())
+    {
+        services.AddHttpsRedirection(options =>
+        {
+            options.RedirectStatusCode = StatusCodes.Status308PermanentRedirect;
+            options.HttpsPort = 443;
+        });
+    }
+}
+```
 
-**Key**: https_port  
-**Type**: *string*  
-**Default**: A default value isn't set.  
-**Set using**: `UseSetting`  
-**Environment variable**: `<PREFIX_>HTTPS_PORT` (The prefix is `ASPNETCORE_` when using the Web Host.)
+When configuring an `IWebHost` in *Program.cs*:
 
 ```csharp
-WebHost.CreateDefaultBuilder(args)
-    .UseSetting("https_port", "8080")
+public static IWebHostBuilder BuildWebHostBuilder(string[] args) =>
```

Comment:
I don't see the need to show both the Startup and WebHost approaches.

Reply:
I'll 🔪 it, but I had it here for newbies and those who prefer the pattern.